### PR TITLE
Fix issue in Java codegen where selectors returned ImmutableMapBuilder instances instead of Map

### PR DESCRIPTION
**Pull request by anderssandven**
Added missing build statement to generated code. This issue caused JsonWriter.writeAny to fail as it doesnt handle the case where we pass in an ImmutableMapBuilder.

### CODE REVIEW

**Comment by BoD** [comment]
Thanks a lot for this! 🙏 I've pushed the .`expected` + one little repro test

File: tests/integration-tests/src/commonTest/kotlin/test/OtherCacheTest.kt [hunk 1]
```diff
@@ -182,4 +183,21 @@ class OtherCacheTest {
 
     assertEquals(instant, response.data!!.instant)
   }
+
+  @Test
+  fun mutationWithoutVariable() = runTest(before = { setUp() }, after = { tearDown() }) {
```
Comment (martinbonnin) [inline tests/integration-tests/src/commonTest/kotlin/test/OtherCacheTest.kt:188]: Nit

```suggestion
  fun mutationWithObjectValueArgument() = runTest(before = { setUp() }, after = { tearDown() }) {
```

**Review by martinbonnin — APPROVED** [review]
Nice 🤩 . Thanks a lot 💙

File: tests/integration-tests/src/commonTest/kotlin/test/OtherCacheTest.kt [hunk 2]
```diff
@@ -182,4 +183,21 @@ class OtherCacheTest {
 
     assertEquals(instant, response.data!!.instant)
   }
+
+  @Test
+  fun mutationWithObjectValueArgument() = runTest(before = { setUp() }, after = { tearDown() }) {
```
Comment (martinbonnin) [inline tests/integration-tests/src/commonTest/kotlin/test/OtherCacheTest.kt:188]: Even more nitpicking (sorry 😅 ). It doesn't have to be a mutation but it has to be using the cache so after some more thoughts, I'd go with:

```suggestion
  fun cacheFieldWithObjectValueArgument() = runTest(before = { setUp() }, after = { tearDown() }) {
```

**Review by BoD — APPROVED** [review]

**Pull request merged**